components/schema-migrator/migrations/director/202002071200_add_package_definitions.up.sql

```diff
@@ -0,0 +1,49 @@
+CREATE TABLE package_definitions (
+    id uuid PRIMARY KEY CHECK (id <> '00000000-0000-0000-0000-000000000000'),
+    tenant_id uuid NOT NULL,
+    name varchar(256) NOT NULL,
+    description text,
+    auth_request_json_schema jsonb,
+    default_auth jsonb
+);
+
+CREATE INDEX ON package_definitions (tenant_id);
+CREATE UNIQUE INDEX ON package_definitions (tenant_id, id);
+
+CREATE TYPE api_instance_auth_status_condition AS ENUM (
+    'PENDING',
+    'SUCCEEDED',
+    'FAILED'
+);
+
+CREATE TABLE api_instance_auths (
+    id uuid PRIMARY KEY CHECK (id <> '00000000-0000-0000-0000-000000000000'),
+    package_id uuid NOT NULL CHECK (id <> '00000000-0000-0000-0000-000000000000'),
+    tenant_id uuid NOT NULL,
+    FOREIGN KEY (tenant_id, package_id) REFERENCES package_definitions (tenant_id, id) ON DELETE CASCADE,
+    context jsonb,
+    auth_value jsonb,
+    status_condition api_instance_auth_status_condition NOT NULL,
+    status_timestamp timestamp NOT NULL,
+    status_message text NOT NULL,
+    status_reason text NOT NULL
+);
+
+CREATE INDEX ON api_instance_auths (tenant_id);
+CREATE UNIQUE INDEX ON api_instance_auths (tenant_id, id);
+CREATE UNIQUE INDEX ON api_instance_auths (tenant_id, package_id, id);
+
+ALTER TABLE api_definitions ADD COLUMN package_definition_id uuid;
```

Comment:
Please also make `app_id` fields for API, Event Definitions and Documents as nullable - for new entries we will use just (tenant_id, package_id) reference